### PR TITLE
Compile benchmarks for Scala 3

### PR DESCRIPTION
**Pull request by kyri-petrou**
I was running some performance tests for Scala 2 vs 3 the other day and I thought you might be interested in being able to run the benchmarks for Scala 3 as well

### CODE REVIEW

**Comment by kyri-petrou** [comment]
I've also updated the Scala versions in the `./sbt` file hoping it's going to make a difference with the flaky test (it failed again)

File: benchmarks/src/main/scala/caliban/GraphQLBenchmarks.scala [hunk 1]
```diff
@@ -361,8 +369,7 @@ class GraphQLBenchmarks {
           new NoUndefinedVariables,
           new NoUnusedFragments,
           new NoUnusedVariables,
-          // new OverlappingFieldsCanBeMerged,
-          new experimental.OverlappingFieldsCanBeMerged,
+          new OverlappingFieldsCanBeMerged,
```
Comment (ghostdogpr) [inline benchmarks/src/main/scala/caliban/GraphQLBenchmarks.scala:372]: Is the experimental one the new default?

**Review by ghostdogpr — APPROVED** [review]

File: benchmarks/src/main/scala/caliban/GraphQLBenchmarks.scala (hunk 1 above)
Comment (kyri-petrou) [reply benchmarks/src/main/scala/caliban/GraphQLBenchmarks.scala:372]: I believe so, `experimental.OverlappingFieldsCanBeMerged` doesn't exist anymore (was probably removed in Sangria 3.x)

Comment (ghostdogpr) [reply benchmarks/src/main/scala/caliban/GraphQLBenchmarks.scala:372]: Thanks!

**Pull request merged**